lib/plugins/blocks.js

```diff
@@ -356,11 +356,12 @@ function inject (bot, { version, storageBuilder }) {
     for (let i = 0; i < packet.records.length; ++i) {
       const record = packet.records[i]
 
-      const blockZ = (record.horizontalPos & 0x0f)
-      const blockX = (record.horizontalPos >> 4) & 0x0f
+      const blockZ = record.horizontalPos !== undefined ? (record.horizontalPos & 0x0f) : record & 0x0f
```

Comment:
can you simplify this by doing 2 blocks with a single if checking what version we are in ?
second point is
> Each entry is composed of the block id, shifted right by 12, and the relative block position in the chunk section (4 bits for x, z, and y, from left to right).

The order is x, z then y